File: DSA/LeetCode/BFS/2583. Kth Largest Sum in a Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthLargestLevelSum(self, root: Optional[TreeNode], k: int) -> int:

        temp = []

        def bfs(node, l):
            if  node is None:
                return -1

            if len(temp) > l:
                temp[l] += node.val
            else:
                temp.append( node.val ) 

            bfs(node.left, l + 1)
            bfs(node.right, l + 1)

        bfs(root, 0)
        temp.sort(reverse=True)
        
        return temp[k - 1] if len(temp) >= k else -1

File: DSA/LeetCode/BFS/2583. Kth Largest Sum in a Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
var kthLargestLevelSum = function (root, k) {



    function bfs(node, l) {
        if (!node)
            return

        if (temp[l] ?? null) {
            temp[l] += node.val
        } else {
            temp[l] = node.val
        }
        bfs(node.left, l + 1)
        bfs(node.right, l + 1)
    }
    let temp = []

    if (!root)
        return -1

    bfs(root, 0)

    temp.sort((a, b) => b - a)

    return temp[k - 1] ?? -1
};